documentation/jetty-documentation/src/main/asciidoc/operations-guide/jaspi/chapter.adoc

```diff
@@ -27,28 +27,29 @@ Only modules conforming to the "Servlet Container Profile" with the ServerAuthMo
 Enable the `jaspi` module:
 
 ----
-include::{JETTY_HOME}/modules/ee10-jaspi.mod[]
+include::{JETTY_HOME}/modules/{ee-current}-jaspi.mod[]
 ----
 
 [[og-jaspi-xml]]
 ===== Configure JASPI
 
-Activate the `ee{9,10}-jaspi` module that matches your EE platform version.
+Activate either the `ee9-jaspi` or `{ee-current}-jaspi` module, whichever matches your EE platform version.
 
+[subs=attributes]
```

Comment:
You're right: this is unnecessary. I had added it in an attempt to fix a separate issue, but that issue turned out to be unrelated. I'll revert.

Reply:
For reasons that weren't totally clear to me, the `[subs=attributes]` annotation was necessary here but not in the other cases.

After a little research, I think I found the explanation [here](https://docs.asciidoctor.org/asciidoc/latest/directives/include/#include-processing):

> Although the include directive looks like a block macro, it’s not a macro and therefore isn’t processed like one. It’s a preprocessor directive; it’s important to understand the distinction.
> 
> A preprocessor directive is processed when the lines of a document are read, but before the document structure is parsed. Therefore, it’s not aware of the surrounding document structure. A preprocessor directive merely adds lines to the reader or takes lines away. The include directive is a preprocessor directive that always adds lines.

So I think you typically do need `[subs=attributes]` for literal blocks that are offset with `----`, but the `include::` directive is a special case, because it is being processed before the document parser runs. Fortunately, it seems like the preprocessor is smart enough to do the attribute substitution, which is why the `include::` directive still executes correctly.